pkg/receive/handler.go

```diff
@@ -375,15 +389,15 @@ func (h *Handler) parallelizeRequests(ctx context.Context, tenant string, replic
 						err = errors.New(errs.Error())
 					}
 				}
-				if err != nil {
-					level.Error(h.logger).Log("msg", "storing locally", "err", err, "endpoint", endpoint)
-				}
-				ec <- err
+				ec <- errors.Wrapf(err, "storing locally, endpoint %v", endpoint)
```

Comment:
I think we should just return if no error?

Reply:
I think I think you rely on `err != nil ` on caller side, maybe I would do it here for readability, but not a blocker.

Reply:
`errors.Wrap` returns nil if the input err is nil